### PR TITLE
Allowing users to pass ggplot to be logged as image

### PR DESCRIPTION
**Pull request by heemanshusuri**

### CODE REVIEW

File: R/run.R [hunk 1]
```diff
@@ -191,6 +191,11 @@ log_image_to_run <- function(name, path = NULL, plot = NULL,
   if (is.null(run)) {
     run <- get_current_run()
   }
+  if (!is.null(plot)) {
+    path <- "_generated_rplot.png"
```
Comment (ninghu) [inline R/run.R]: it should be created under system standard temp directory and get cleaned up at the end.

File: R/run.R [hunk 2]
```diff
@@ -191,6 +191,11 @@ log_image_to_run <- function(name, path = NULL, plot = NULL,
...
+    ggsave(filename = path, plot = plot)
+    plot <- NULL
```
Comment (ninghu) [inline R/run.R]: why set plot to NULL?

File: R/run.R [hunk 3]
```diff
@@ -191,6 +191,11 @@ log_image_to_run <- function(name, path = NULL, plot = NULL,
   if (is.null(run)) {
     run <- get_current_run()
   }
+  if (!is.null(plot)) {
+    path <- "_generated_rplot.png"
+    ggsave(filename = path, plot = plot)
```
Comment (ninghu) [inline R/run.R]: is ggsave from ggplot2? if so, shouldn't claim gglot2 as dependency of r sdk?

File: R/run.R [hunk 4]
```diff
@@ -191,6 +191,11 @@ log_image_to_run <- function(name, path = NULL, plot = NULL,
   if (is.null(run)) {
     run <- get_current_run()
   }
+  if (!is.null(plot)) {
```
Comment (ninghu) [inline R/run.R]: update the docstring for plot parameter?

File: R/run.R (hunk 2 above)
Comment (heemanshusuri) [reply R/run.R]: both path and plat can't be non null otherwise python sdk would throw validation error.

File: R/run.R (hunk 3 above)
Comment (heemanshusuri) [reply R/run.R]: yes, added

File: R/run.R (hunk 1 above)
Comment (heemanshusuri) [reply R/run.R]: made the name consistent with python sdk.

File: R/run.R (hunk 4 above)
Comment (heemanshusuri) [reply R/run.R]: done

File: R/run.R [hunk 5]
```diff
@@ -188,15 +188,25 @@ log_confusion_matrix_to_run <- function(name, value, description = "",
 #' @export
 log_image_to_run <- function(name, path = NULL, plot = NULL,
                              description = "", run = NULL) {
+  if (!is.null(path) && !is.null(plot) {
+    stop(paste("Invalid parameters, path and plot were both provided,",
```
Comment (ninghu) [inline R/run.R]: nit: use paste0 if separator is empty.

Comment (heemanshusuri) [reply R/run.R]: done

**Pull request merged**